### PR TITLE
Draft: [5600] - (enhancement/semgrep/test) - modifying the semgrep test output to include missed and incorrect lines

### LINKED ISSUE #5600: Output missed/incorrectly detected line in test rather than whole arrays
**Is your feature request related to a problem? Please describe.**
When a test fails and there are numerous expected lines to be detected, the output is really not helpful and it's hard to quickly see which lines were missed and/or incorrectly detected:

```
	✖ test-rule    expected lines: [4, 7, 10, 13, 17, 20, 23, 26, 29, 32, 35, 38, 41, 44, 47, 51, 57, 59, 63, 67, 70, 73, 76, 78, 80, 83, 85, 87, 90, 93, 125],
reported lines: [4, 7, 10, 13, 17, 20, 23, 26, 29, 32, 35, 38, 41, 47, 51, 57, 59, 63, 67, 70, 73, 76, 78, 80, 83, 85, 87, 90, 93, 125, 158]
```

**Describe the solution you'd like**
Do a diff of the arrays to output the missed lines (False Negative), as well as incorrectly detected (False Positive) ones when the test fails. e.g:

```
	✖ test-rule   Missed Lines: [44]
                      Incorrectly Detected Lines: [158] 
```

**Describe alternatives you've considered**
Checked the help (`semgrep test -h`) but there does not seem to be an option to have such output.

**Use case**
Won't have to spend some time trying to figure out the diff of the `expected lines` and `reported lines` and kill my eyes in the process

### PR DESCRIPTION
**Pull request by vivekkhimani**
Acceptance Criteria (AC):
- Fixes https://github.com/returntocorp/semgrep/issues/5600.

PR checklist:

- [x] Documentation is up-to-date
- [x] `changelog.d/<issue>.<type>` is a file with the _what_, _why_, and _how_ of the change.
  - \<issue> is `pa-312` (Linear ticket), `gh-1234` (GitHub issue), or `new-gizmo` (unique semantic name)
  - \<type> is `added`, `changed`, `fixed`, or `infra`.
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see the [Contribution guidelines](https://semgrep.dev/docs/contributing/how-to-contribute/)!

### CODE REVIEW

File: cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-github-pr/results.txt [hunk 1]
```diff
@@ -47,22 +22,43 @@ Using <MASKED> as the merge-base of <MASKED> and <MASKED>
...
+  -M            detect renames.
+  -C            detect copies.
+  --find-copies-harder
+                try unchanged files as candidate for copy detection.
+  -l<n>         limit rename attempts up to <n> paths.
+  -O<file>      reorder diffs according to the <file>.
+  -S<string>    find filepair whose only one side contains the string.
+  --pickaxe-all
+                show all files diff when -S is used and hit is found.
+  -a  --text    treat all files as text.
```
Comment (underyx) [inline cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-github-pr/results.txt]: these snapshot changes seem unexpected 🤔

Comment (underyx) [reply cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-github-pr/results.txt]: it seems like `git` doesn't run correctly in the tests directory (or the temporary directory the tests create)?

Comment (vivekkhimani) [reply cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-github-pr/results.txt]: yep, I noticed! I didn't notice before pushing it lol. I can revert it or further investigate it. What do you suggest?

Comment (underyx) [reply cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-github-pr/results.txt]: I think you could try running `git status` or something like that manually in a test's temporary directory, and that should tell you why it's failing so you can fix it.

Comment (vivekkhimani) [reply cli/tests/e2e/snapshots/test_ci/test_full_run/autofix-github-pr/results.txt]: Oh yes, I didn't get to this during rest of the weekend so this shouldn't be ready for review yet. I will put it in draft mode. My bad!

**Comment by vivekkhimani** [comment]
Don't think this was merged but definitely closed as I am starting it from scratch. Deleting the branch.

**Pull request closed without merging**